File: sdks/thunkmetrc-py/wrapper/src/thunkmetrc/wrapper/services/units_of_measure_service.py
from typing import Any, Optional, List, AsyncGenerator
from thunkmetrc.client import MetrcClient
from ..ratelimiter import MetrcRateLimiter
from ..models import *
from ..utils import iterate_pages, deserialize_page

class UnitsOfMeasureService:
    def __init__(self, client: MetrcClient, limiter: MetrcRateLimiter):
        self.client = client
        self.limiter = limiter

    async def get_active_units_of_measure(self, body: Any = None) -> PaginatedResponse['UnitOfMeasure']:
        """
        Retrieves all active units of measure.
        GET /unitsofmeasure/v2/active
        Parameters:
        """
        async def op():
            return await self.client.units_of_measure.get_active_units_of_measure(body=body,
            )
        return await self.limiter.execute(None, True, op)

    async def get_inactive_units_of_measure(self, body: Any = None) -> PaginatedResponse['UnitOfMeasure']:
        """
        Retrieves all inactive units of measure.
        GET /unitsofmeasure/v2/inactive
        Parameters:
        """
        async def op():
            return await self.client.units_of_measure.get_inactive_units_of_measure(body=body,
            )
        return await self.limiter.execute(None, True, op)

File: sdks/thunkmetrc-ts/wrapper/src/services/UnitsOfMeasureService.ts

import { MetrcClient as InternalMetrcClient } from '@thunkier/thunkmetrc-client';
import { MetrcRateLimiter } from '../RateLimiter';
import { iteratePages } from '../Utils';
import { 
    PaginatedResponse,
    UnitOfMeasure,
} from '../models';

export class UnitsOfMeasureService {
  constructor(private client: InternalMetrcClient, private rateLimiter: MetrcRateLimiter) {}

  /**
   * Retrieves all active units of measure.
   * GET GetActiveUnitsOfMeasure
   */
  public async getActiveUnitsOfMeasure(body?: any): Promise<PaginatedResponse<UnitOfMeasure>> {
    return this.rateLimiter.execute(null,true,
        () => this.client.getActiveUnitsOfMeasure(body,
        )
    );
  }

  /**
   * Retrieves all inactive units of measure.
   * GET GetInactiveUnitsOfMeasure
   */
  public async getInactiveUnitsOfMeasure(body?: any): Promise<PaginatedResponse<UnitOfMeasure>> {
    return this.rateLimiter.execute(null,true,
        () => this.client.getInactiveUnitsOfMeasure(body,
        )
    );
  }
}

